_search-plugins/sql/ppl/functions.md

```diff
@@ -771,3 +771,198 @@ sepal_length_in_cm | sepal_width_in_cm | petal_length_in_cm | petal_width_in_cm
 | 5.1 | 3.5 | 1.4 | 0.2 | 1   
 | 5.6 | 3.0 | 4.1 | 1.3 | 0
 | 6.7 | 2.5 | 5.8 | 1.8 | 2
+
+## join
+
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, join the discussion on the [OpenSearch forum](https://forum.opensearch.org/).    
+{: .warning}
+
+You can combine two datasets using the `join` command. The left side can be an index or results from piped commands, while the right side can be either an index or a subquery.
+
+### Syntax
+
+```sql
+[join-type] join [left-alias] [right-alias] on <join-criteria> <right-dataset>
+* joinCriteria: mandatory. It could be any comparison expression.
+* right-dataset: mandatory. Right dataset could be either an index or a subquery with/without alias.
+```
+
+Field | Description | Type | Required | Default
+:--- |:--- | :--- | :--- | :---
+`join-type` | The type of join to perform. Valid values are `inner`, `left`, `right`, `full`, `cross`, `semi`, and `anti`.  | `String` | No   | `inner`
+`left-alias` | The subquery alias to use with the left join side in order to avoid ambiguous naming. Fixed pattern: `left = <left-alias>`    | `String` | No   |
+`right-alias` | The subquery alias to use with the right join side in order to avoid ambiguous naming. Fixed pattern: `right = <right-alias>` | `String` | No   |
```

Comment:
```suggestion
`right-alias` | The subquery alias to use with the right join side in order to avoid ambiguous naming. Fixed pattern: `right = <right-alias>` | `String` | No   | N/A
```